src/javascript/controller/weatherControler.js

```diff
@@ -0,0 +1,96 @@
+import changeColor from '../helpers/updateColor';
+import uppercaseFirstLetter from '../helpers/uppercaseFirstLetter';
+import fetchImageData from '../model/imageData';
+
+import fetchWeatherData from '../model/weatherData';
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function updateBodyBackground(queryText) {
+  const [returnedImage, avarageColorCode] = await fetchImageData(queryText);
+
+  const body = document.querySelector('body');
+  body.style.backgroundImage = `url(${returnedImage})`;
+  body.style.backgroundSize = 'cover';
+
+  changeColor(avarageColorCode);
+}
+
+const populateMain = () => {
+  const locationInput = document.getElementById('location-input');
+  const resultView = document.getElementsByClassName('result-view')[0];
+  const locationNameView = document.getElementById('location-name');
+  const temperateView = document.createElement('h1');
+  const searchButton = document.getElementById('search-button');
+  const weatherIcon = document.getElementById('weather-icon');
+  const weatherDescription = document.getElementById('weather-desciption');
+  const unitChanger = document.getElementById('unit-changer');
+
+  let tempInC;
+  let tempInF;
+  let weatherCondition;
+  let countryAbbr;
+
+  const setTemperatureView = () => {
+    if (tempInF === undefined) {
+      temperateView.innerText = 'You must first check weather for city!';
+    } else if (unitChanger.checked === true) {
+      temperateView.innerText = `${(Math.round(tempInF * 100) / 100).toString()} °F`;
+    } else if (unitChanger.checked === false) {
+      temperateView.innerText = `${(Math.round(tempInC * 100) / 100).toString()} °C`;
+    }
+
+    if (locationInput.value !== '') {
+      locationNameView.innerText = `${uppercaseFirstLetter(locationInput.value)}, ${countryAbbr}`;
+    }
+    weatherDescription.innerText = weatherCondition.description;
+    weatherIcon.src = `http://openweathermap.org/img/w/${weatherCondition.icon.toString()}.png`;
+    resultView.appendChild(temperateView);
+
+    locationInput.value = '';
+  };
+
+  const setMainView = async () => {
+    locationNameView.innerHTML = '<div class="lds-ellipsis"><div></div><div></div><div></div><div></div></div>';
+    const weatherData = await fetchWeatherData(locationInput.value);
+    if (locationInput.value === '') {
+      locationNameView.innerText = 'You must enter a location name first!';
+    } else if (weatherData === null) {
+      locationNameView.innerText = 'The location couldnt found!';
+    } else {
+      // [, temperateView.innerText] = Math.round(temperatures);
```

Comment:
### Remove commented out code
- [ ] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Reply:
Done.